cezo_fl/server.py

```diff
@@ -122,19 +101,26 @@ def __init__(
         self.seed_grad_records = SeedAndGradientRecords()
         self.client_last_updates = [0 for _ in range(len(self.clients))]
 
-        self.server_model: torch.nn.Module | None = None
-        self.server_criterion: torch.nn.Module | None = None
-        self.optim: torch.optim.Optimizer | None = None
+        self.server_model: Optional[torch.nn.Module] = None
```

Comment:
which python version you used? Now it is preferred to use `x | None` than `Optional[x]`

Reply:
I am using python3.10.14 + black for formatting + flake8 for linting. Do not get a warning for using Optional, Also I do not find a strong opinion today googling which style is preferred. 
For me when reading code, Optional means I can skip this variable by just reading the type, while ' | None' makes me think for a bit before I can realize it's optional.
So I prefer keeping Optional, unless your linter/formatter gives direct warning when using it.

Reply:
Decided to go with x | None